### ISSUE TITLE
Is it wrong for these code?

### ISSUE THREAD

**Issue by luocanrao** [r9y9/wavenet_vocoder#38]
y_hat = model(x, c=c, g=g, softmax=False)

    if is_mulaw_quantize(hparams.input_type):
        # wee need 4d inputs for spatial cross entropy loss
        # (B, C, T, 1)
        y_hat = y_hat.unsqueeze(-1)
        loss = criterion(y_hat[:, :, :-1, :], y[:, 1:, :], mask=mask) # criterion  is nn.CrossEntropyLoss
    else:
        loss = criterion(y_hat[:, :, :-1], y[:, 1:, :], mask=mask)

I am confuzing about this code, why y_hat'shape is ( B, C, T, 1), but y'shape is (B, T, 1),  but why y_hat'shape is not (B, T, 1, C). 
I read pytorch's docs about CrossEntropyLoss function.

class torch.nn.CrossEntropyLoss(weight=None, size_average=True, ignore_index=-100, reduce=True)
Input: (N,C) where C = number of classes
Target: (N) where each value is  0≤targets[i]≤C−1
Examples:
>>> loss = nn.CrossEntropyLoss()
>>> input = torch.randn(3, 5, requires_grad=True)
>>> target = torch.LongTensor(3).random_(5)
>>> output = loss(input, target)
>>> output.backward()

can somebody give me some help?

**Comment by r9y9** [comment]
See docs for `cross_entropy2d` https://github.com/pytorch/pytorch/pull/687/files. The spatial case of cross entropy is not well undocumented at the moment (see https://github.com/pytorch/pytorch/pull/687#issuecomment-295743429).
